packages/complex-matcher/src/index.js

```diff
@@ -599,33 +599,45 @@ export const parse = parser.parse.bind(parser)
 
 // -------------------------------------------------------------------
 
-const _getPropertyClauseStrings = ({ child }) => {
+const _getPropertyClauseStringsOrRegex = ({ child }) => {
   if (child instanceof Or) {
     const strings = []
     child.children.forEach(child => {
       if (child instanceof StringNode) {
         strings.push(child.value)
       }
+      if (child instanceof RegExpNode) {
+        const unescapedRegexp = child.re.source.replace(/^(\^)|\\|\$$/g, '')
+        if (child.re.source === `^${escapeRegExp(unescapedRegexp)}$`) {
```

Comment:
Move into a dedicated function (maybe `extractStringFromRegexp` which would return false if not possible).